squad/ci/backend/lava.py

```diff
@@ -220,25 +220,28 @@ def __download_full_log__(self, job_id):
     def __download_test_log__(self, raw_log, log_start, log_end):
         if not log_start:
             return ""
-        return_lines = StringIO()
-        log_start_line = int(log_start)
-        log_end_line = None
-        if log_end:
-            log_end_line = int(log_end)
-        else:
-            log_end_line = log_start_line + 2  # LAVA sometimes misses the signals
-        raw_log.seek(0)
-        counter = 0
-        for line in raw_log:
-            counter += 1
-            if counter < log_start_line:
-                continue
-            return_lines.write(line.decode("utf-8"))
-            return_lines.write("\n")
-            if counter >= log_end_line:
-                break
-        raw_log.seek(0)
-        return return_lines.getvalue()
+        try:
+            return_lines = StringIO()
+            log_start_line = int(log_start)
+            log_end_line = None
+            if log_end:
+                log_end_line = int(log_end)
+            else:
+                log_end_line = log_start_line + 2  # LAVA sometimes misses the signals
+            raw_log.seek(0)
+            counter = 0
+            for line in raw_log:
+                counter += 1
+                if counter < log_start_line:
+                    continue
+                return_lines.write(line.decode("utf-8"))
+                return_lines.write("\n")
+                if counter >= log_end_line:
+                    break
+            raw_log.seek(0)
+            return return_lines.getvalue()
+        except UnicodeDecodeError as e:
+            return 'Failed getting test log: %s' % (str(e))
```

Comment:
we shouldn't need to throw away the entire log because a single line fails to decode. there are workarounds: for example you can use something like this:

```python
        try:
            line = line.decode('utf-8')
        except UnicodeDecodeError:
            line = line.decode('iso-8859-1')
```

Reply:
hum, didn't think of trying to use another table. will fix